pkg/cloud/aws/actuators/scope.go

```diff
@@ -59,17 +65,25 @@ func NewScope(params ScopeParams) (*Scope, error) {
 		return nil, errors.Errorf("failed to load cluster provider status: %v", err)
 	}
 
-	session, err := session.NewSession(aws.NewConfig().WithRegion(clusterConfig.Region))
-	if err != nil {
-		return nil, errors.Errorf("failed to create aws session: %v", err)
+	sessionLock.Lock()
+	defer sessionLock.Unlock()
+
+	var localSession *session.Session
```

Comment:
It might be cleaner to move this to a function - something like `sessionForRegion(region)`.